src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,125 @@
-import React from 'react';
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import classNames from 'classnames';
+import { Comment, CommentData } from '../types/Comment';
+import { createComment } from '../api/comments';
+import { warningTimer } from '../utils/warningTimer';
+
+type Props = {
+  postId: number | undefined;
+  setComments: Dispatch<SetStateAction<Comment[]>>
+  setIsHasError: Dispatch<SetStateAction<boolean>>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  setComments,
+  setIsHasError,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const DefaultError = {
+    errorName: false,
+    errorEmail: false,
+    errorBody: false,
+  };
+  const nameError = 'errorName';
+  const emailError = 'errorEmail';
+  const bodyError = 'errorBody';
+  const [isError, setIsError] = useState(DefaultError);
```

Comment:
actually, you aren't saving a boolean value in this state, but object, so rename it according to it's purpose.
```suggestion
  const [errors, setErrors] = useState(DefaultErrors);
```